src/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react'
+
+import styles from './Sidebar.module.css'
+
+import { MOBILE_WIDTH } from '../../shared/constants'
+import { useDispatch, useSelector } from 'react-redux'
+import {
+    areFiltersVisible,
+    setCategory,
+    toggleProvider,
+} from '../../redux/actions/filters'
+
+function Sidebar() {
+    const dispatch = useDispatch()
+
+    const [recentlySelected, setRecentlySelected] = useState([])
+
+    const filtersVisibility = useSelector(
+        state => state.filters.areFiltersVisible
+    )
+    const width = useSelector(state => state.data.width)
+    const categories = useSelector(state => state.data.data.categories)
+    const providers = useSelector(state => state.data.data.providers)
+    const selectedCategory = useSelector(
+        state => state.filters.selectedCategory
+    )
+    const selectedProviders = useSelector(
+        state => state.filters.selectedProviders
+    )
+
+    const changeFiltersVisibility = () =>
+        dispatch(areFiltersVisible(!filtersVisibility))
+
+    const select = (id, isCategory) => {
+        isCategory ? dispatch(setCategory(id)) : dispatch(toggleProvider(id))
+
+        if (width === MOBILE_WIDTH)
+            setRecentlySelected([...recentlySelected, { id, isCategory }])
+    }
+
+    const cancel = () => {
+        recentlySelected.map(({ id, isCategory }) => {
+            isCategory
+                ? dispatch(setCategory(null))
+                : dispatch(toggleProvider(id))
+        })
+
+        changeFiltersVisibility()
+        setRecentlySelected([])
+    }
+
+    const applyFilters = () => {
+        changeFiltersVisibility()
+        setRecentlySelected([])
+    }
+
+    const renderAllCategories = () =>
+        categories.map(category => (
+            <div
+                key={category.id}
+                className={`${styles['category']} ${
+                    category.id === selectedCategory ? styles['selected'] : ''
+                }`}
+                onClick={select.bind(this, category.id, true)}
+            >
+                <img src={category.icon} alt="category-icon" />
+                <p>{category.displayName}</p>
+            </div>
+        ))
+
+    const renderAllProviders = () =>
```

Comment:
why do we need this? shouldn't the list be re-rendered whenever we have some changes in store? we can just put this `.map` to the `return` and that should be enough. Am I missing something?

Reply:
Discussed this with Andrii Halich, he said it wouldn`t harm the productivity and the speed of an app and suggested to move some logic to different components, so created a separate component for Categories, Providers and Filter button